### DISCUSSION TITLE
Jolokia 2 JBOSS metris

### DISCUSSION THREAD

**Discussion by dumpap** [rhuss/jolokia#718] [Q&A]
Hi Team,

I have a Jolokia 2 agent attached two 2 java processes (apps running in JBOSS) in windows server. They are on two ports.

I have a telegraf agent which uses endpoints on above 2 ports to get Java metrics and I can see that the metrics are being captured. 

I am trying to get the JBOSS metrics with the below configuration. But some have all the metrics are coming as 0 value.  

Can someone please shed some light on this ? 

### JBOSS

  [[inputs.jolokia2_agent.metric]]
    name     = "connectors.http"
    mbean    = "jboss.as:https-listener=*,server=*,subsystem=undertow"
    paths    = ["bytesReceived","bytesSent","errorCount","requestCount"]
    tag_keys = ["server","https-listener"]

  [[inputs.jolokia2_agent.metric]]
    name     = "connectors.http"
    mbean    = "jboss.as:http-listener=*,server=*,subsystem=undertow"
    paths    = ["bytesReceived","bytesSent","errorCount","requestCount"]
    tag_keys = ["server","http-listener"]

  [[inputs.jolokia2_agent.metric]]
    name     = "datasource.jdbc"
    mbean    = "jboss.as:data-source=*,statistics=jdbc,subsystem=datasources"
    paths    = ["PreparedStatementCacheAccessCount","PreparedStatementCacheHitCount","PreparedStatementCacheMissCount"]
    tag_keys = ["data-source"]

  [[inputs.jolokia2_agent.metric]]
    name     = "datasource.pool"
    mbean    = "jboss.as:data-source=*,statistics=pool,subsystem=datasources"
    paths    = ["AvailableCount","ActiveCount","MaxUsedCount"]
    tag_keys = ["data-source"]

Regards,
Pradeep

**Comment by grgrzybek** [comment]
What's this kind of syntax? Did you try with `curl`?

**Reply by dumpap** [reply]
this is the Telegraf configuration file, which make calls to Jolokia on assigned port to capture metrics. What the issue I am facing is all the metrics for JBOSS are coming as 0 value.

**Comment by grgrzybek** [comment]
We'd need to verify the values by checking other JMX client (remote JMX, jconsole, visualvm, ...). Without this data I can't help much...
